src/plugins/mvcc_delete_plugin.cpp

```diff
@@ -0,0 +1,177 @@
+      DebugAssert(chunk_id < (table->chunk_count() - 1),
+                  "MVCC Logical Delete should not be applied on the last/current mutable chunk.")
+
+      // Create temporary referencing table that contains the given chunk only
+      auto transaction_context = TransactionManager::get().new_transaction_context();
+  auto gt = std::make_shared<GetTable>(table_name);
+  gt->set_transaction_context(transaction_context);
+
+  auto chunk_count = table->chunk_count();
+  std::vector<ChunkID> excluded_chunk_ids;
+  for (ChunkID excluded_chunk_id = ChunkID{0}; excluded_chunk_id < chunk_count; ++excluded_chunk_id) {
+    if (chunk_id != excluded_chunk_id) {
+      excluded_chunk_ids.push_back(excluded_chunk_id);
+    }
+  }
+
+  gt->set_excluded_chunk_ids(excluded_chunk_ids);
+  gt->execute();
+
+  // Validate temporary table
+  auto validate_table = std::make_shared<Validate>(gt);
+  validate_table->set_transaction_context(transaction_context);
+  validate_table->execute();
+
+  // Use Update operator to delete and re-insert valid records in chunk
+  // Pass validate_table into Update operator twice since data will not be changed.
+  auto update_table = std::make_shared<Update>(table_name, validate_table, validate_table);
+  update_table->set_transaction_context(transaction_context);
+  update_table->execute();
+
+  // Check for success
+  if (update_table->execute_failed()) {
+    transaction_context->rollback();
+    return false;
+  } else {
+    // TODO(all): Check for success of commit, currently (2019-01-11) not yet possible.
+    transaction_context->commit();
+
+    // Mark chunk as logically deleted
+    chunk->set_cleanup_commit_id(transaction_context->commit_id());
+    std::cout << "Deleted chunk " << chunk_id << " logically." << std::endl;
+    return true;
+  }
+}
+
+bool MvccDeletePlugin::_delete_chunk_physically(const std::string& table_name, const ChunkID chunk_id) {
+  const auto& table = StorageManager::get().get_table(table_name);
+
+  DebugAssert(table->get_chunk(chunk_id) != nullptr, "Chunk does not exist. Physical Delete can not be applied.")
+
+      // Check whether there are still active transactions that might use the chunk
+      CommitID cleanup_commit_id = table->get_chunk(chunk_id)->get_cleanup_commit_id();
+  CommitID lowest_snapshot_commit_id = TransactionManager::get().get_lowest_active_snapshot_commit_id();
+  if (cleanup_commit_id < lowest_snapshot_commit_id) {
+    DebugAssert(table->chunks()[chunk_id].use_count() == 1,
+                "At this point, the chunk should be referenced by the "
+                "Table-chunk-vector only.")
+        // Usage checks have been passed. Apply physical delete now.
+        table->remove_chunk(chunk_id);
+    std::cout << "Deleted chunk " << chunk_id << " physically." << std::endl;
+    return true;
+  } else {
+    // Chunk might still be in use. Wait with physical delete.
+    return false;
```

Comment:
A method `_delete_chunk_physically` should not decide if it deletes the chunk or not. Either call it `_try_...` or extract the part that checks if it can be deleted.